lib/API/JsonApi/v1/Filters/Interpreters/ByPreMadeFilters/Today.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ *
+ * SugarCRM Community Edition is a customer relationship management program developed by
+ * SugarCRM, Inc. Copyright (C) 2004-2013 SugarCRM Inc.
+ *
+ * SuiteCRM is an extension to SugarCRM Community Edition developed by SalesAgility Ltd.
+ * Copyright (C) 2011 - 2017 SalesAgility Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License version 3 as published by the
+ * Free Software Foundation with the addition of the following permission added
+ * to Section 15 as permitted in Section 7(a): FOR ANY PART OF THE COVERED WORK
+ * IN WHICH THE COPYRIGHT IS OWNED BY SUGARCRM, SUGARCRM DISCLAIMS THE WARRANTY
+ * OF NON INFRINGEMENT OF THIRD PARTY RIGHTS.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+namespace SuiteCRM\API\JsonApi\v1\Filters\Interpreters\ByPreMadeFilters;
+
+use SuiteCRM\API\JsonApi\v1\Filters\Interfaces\ByPreMadeFilterInterpreter;
+
+class Today implements ByPreMadeFilterInterpreter
+{
+
+    /**
+     * @param string $name
+     * @return bool
+     */
+    public function hasByPreMadeFilter($name)
```

Comment:
Is this really necessary?

Reply:
yes.

Reply:
Since you only return one value, you can use a constant

Reply:
Why? Actually you use boolean when you check if it's 'Today' or not.

[if($interpreter->hasByPreMadeFilter($filterName)) {}]

You can use something like this instead

const TODAY = 'Today'

then when you call

if($filterName === Today::TODAY){}

Reply:
hasPreMadeFilter enables it's callers to has if this filter is matches the filter they are looking for. 

Since we don't want the callers to have to understand how the interpreters or operators work. That is the responsibility of the this class to make this decision, and to handle any related behavior.

I don't think the const is really not necessary here. By making something const you are exposing the inner workings or state of this class. 

Why do we need to write extra lines of code for a simple check?

Reply:
I don't agree. You only use it at one line, but fine.